schema_salad/avro/schema.py

```diff
@@ -740,6 +781,11 @@ def make_avsc_object(json_data: JsonDataType, names: Optional[Names] = None) ->
         if atype in VALID_TYPES:
             if atype == "array":
                 items = json_data.get("items")
+                if "name" in json_data and json_data["name"]:
```

Comment:
Checking that `json_data["name"]` is not None nor that it is an empty string is a requirement, as otherwise the metaschema fails to load (for some reason the ArraySchemas there try to load with empty `name` strings).